File: continuum-core/src/main/java/org/kinotic/continuum/api/security/Participant.java
package org.kinotic.continuum.api.security;

import org.kinotic.continuum.api.Identifiable;

import java.util.List;
import java.util.Map;

/**
 * Created by Navíd Mitchell 🤪on 6/16/23.
 */
public interface Participant extends Identifiable<String> {
    /**
     * The identity of the participant
     *
     * @return the identity of the participant
     */
    @Override
    String getId();

    /**
     * The tenant that the participant belongs to
     *
     * @return the tenant or null if not using multi-tenancy
     */
    String getTenantId();

    /**
     * Metadata is a map of key value pairs that can be used to store additional information about a participant
     *
     * @return a map of key value pairs
     */
    Map<String, String> getMetadata();

    /**
     * Roles are a list of strings that can be used to authorize a participant to perform certain actions
     *
     * @return a list of roles
     */
    List<String> getRoles();
}

File: continuum-js/continuum-client/src/api/security/Participant.ts
import { IParticipant } from "./IParticipant";

/**
 * Created by Navid Mitchell on 6/2/20
 */
export class Participant implements IParticipant {

    public id: string;

    public tenantId?: string | null;

    public metadata: Map<string, string>;

    public roles: string[];

    constructor(id: string,
                tenantId?: string,
                metadata?: Map<string, string>,
                roles?: string[]) {
        this.id = id
        this.tenantId = tenantId;
        this.metadata = metadata || new Map();
        this.roles = roles || [];
    }

}
